### PR TITLE
Improve actor telemetry types

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Improve actor telemetry, add type name override property
* Make actor telemetry message constructors public

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: src/core/Akka/Actor/ActorCell.FaultHandling.cs [hunk 1]
```diff
@@ -373,6 +373,13 @@ private void FinishRecreate(Exception cause, ActorBase failedActor)
 
         }
 
+        /// <summary>
+        /// Overrideable in order to support issues such as https://github.com/petabridge/phobos-issues/issues/82
+        /// </summary>
+        protected virtual ActorRestarted CreateActorRestartedEvent(Exception cause)
```
Comment (Arkatufus) [inline src/core/Akka/Actor/ActorCell.FaultHandling.cs:379]: Add override for ActorRestarted

File: src/core/Akka/Actor/ActorTelemetry.cs [hunk 2]
```diff
@@ -30,50 +31,67 @@ public interface IActorTelemetryEvent : INoSerializationVerificationNeeded, INot
         /// The implementation type for this actor.
         /// </summary>
         Type ActorType { get; }
+        
+        /// <summary>
+        /// A type name override for the actor
+        /// </summary>
+        public string ActorTypeOverride { get; }
```
Comment (Arkatufus) [inline src/core/Akka/Actor/ActorTelemetry.cs:38]: Add new actor type name override property

File: src/core/Akka/Actor/ActorTelemetry.cs [hunk 3]
```diff
@@ -30,50 +31,67 @@ public interface IActorTelemetryEvent : INoSerializationVerificationNeeded, INot
...
     }
     
     // Create ActorTelemetryEvent messages for the following events: starting an actor, stopping an actor, restarting an actor
     public sealed class ActorStarted : IActorTelemetryEvent
     {
-        internal ActorStarted(IActorRef subject, Type actorType)
+        public ActorStarted(IActorRef subject, Type actorType, string? actorTypeOverride = null)
```
Comment (Arkatufus) [inline src/core/Akka/Actor/ActorTelemetry.cs:44]: Make actor telemetry message constructors public

**Review by Arkatufus — COMMENTED** [review]
Self-review

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**